gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/DuplicateObservationStatementMapperTest.java

```diff
@@ -229,37 +244,61 @@ public void doesntMergeObservationWhereTheMatchingObservationHasMultiplePertinen
             generateLinksetComponent("ID-1")
         ));
 
-        mapper.mergeDuplicateObservationStatements(ehrExtract);
+        mapper.mergeOrRemoveDuplicateObservationStatements(ehrExtract);
 
         assertThat(firstEhrComposition(ehrExtract)).hasSize(3);
     }
 
     @Test
-    public void doesntMergeObservationWhereTheObservationDoesNotEndInEllipses() {
+    public void doesntMergeObservationWhereTheObservationDoesNotEndInEllipsesAndCodeableConceptCodesAreDifferent() {
         var ehrExtract = createExtract(List.of(
                 createObservation("ID-1", "101", "This is an observation doesnt end with ellipses:::"),
-                createObservation("ID-2", "101", "This is an observation which ends with ellipses removed."),
+                createObservation("ID-2", "102", "This is an observation which ends with ellipses removed."),
                 generateLinksetComponent("ID-1")
         ));
 
-        mapper.mergeDuplicateObservationStatements(ehrExtract);
+        mapper.mergeOrRemoveDuplicateObservationStatements(ehrExtract);
 
         assertThat(firstEhrComposition(ehrExtract)).hasSize(3);
     }
 
     @Test
-    public void doesntMergeObservationWhereTheObservationIsLessThan47Chars() {
+    public void removeObservationWhereTheObservationDoesNotEndInEllipsesButCodeableConceptCodesAreTheSame() {
```

Comment:
I think this test name doesn't actually match up with the behaviour?

Reply:
the test covers the behaviour in `removeExactDuplicates` method

Reply:
But it doesn't mirror [the requirements you documented within the ticket](https://nhse-dsic.atlassian.net/browse/NIAD-3187?focusedCommentId=105951).

> and one includes a pertinent annotation while the other does not

Reply:
thanks, good catch, I've aligned the code with the acceptance criteria